medusa/essentials/dictionary.py

```diff
@@ -0,0 +1,28 @@
+# coding=utf-8
+"""Extended dictionaries."""
+
+from collections import OrderedDict
+
+
+def not_none(x):
+    """Return true is x is not none."""
+    return x is not None
+
+
+def not_empty(x):
+    """Return true is x is not none and not empty."""
+    return x is not None and x != ''
+
+
+class OrderedPredicateDict(OrderedDict):
+    """Ordered dictionary that only accept keys that matches the predicate."""
```

Comment:
Why does this need to be an OrderedDict? Dicts are better performance so unless maintaining insertion order is absolutely necessary dict should be preferred.